themes/default/content/blog/changes-to-import/index.md

```diff
@@ -0,0 +1,235 @@
+
+At Pulumi, we understand that many cloud engineers and platform teams around the world don't have the luxury of greenfield projects, more often than not we're stuck with the impossible task of "refactoring" or "migrating" existing projects to more modern stacks to help increase team productivity, velocity, and stability. These projects aren't trivial and we want to make it easier for teams and organizations to bring their infrastructure into a cloud engineering world. Oh, and worry not you lucky greenfielders ... even if you just wanna ClickOps your way through some resources and import them into your program; that's gonna work just fine too; we won't tell if you don't 😉
+
+To help you understand the changes we've made to help you on your journey, let's take a look at some side by sides examples of how import used to work vs. what we're releasing today.
+
+### Example: AWS S3 Bucket
+
+Previously when importing an S3 bucket, we would set the default values for properties like `acl` and `forceDestroy`. The new behavior will not include properties with default values in the generated code, keeping your code more inline with how you'd write it yourself. Artisanal codegen for the artisinal cloud engineer.
+
+#### Code Comparison
+
+This code would have been generated by the older implementation:
+
+```typescript
+const my_bucket = new aws.s3.Bucket("my-bucket", {
+    acl: "private",
+    bucket: "my-bucket-3f85c54",
+    forceDestroy: false,
+}, {
+    protect: true,
+});
+```
+
+This code will be generated by the newer implementation:
+
+```typescript
+const my_bucket = new aws.s3.Bucket("my-bucket", {
+    arn: "arn:aws:s3:::my-bucket-3f85c54",
+    bucket: "my-bucket-3f85c54",
+    hostedZoneId: "Z3BJ6K6RIION7M",
+    requestPayer: "BucketOwner",
+}, {
+    protect: true,
+});
+```
+
+The above accurately reflects what the providers `Read` function has told us of the inputs set for this bucket, although due to the way the AWS provider happens to work we could elide all these properties. This is because the provider will use outputs saved in the state file and the new inputs to calculate the update diff, and a property that exists in the old output and is missing from the new inputs is simply left at the old output value. As none of the properties for `Bucket` are marked required the following also works and is equivalent:
+
+```
+const my_bucket = new aws.s3.Bucket("my-bucket", { }, { protect: true });
+```
+
+### Example: AWS EC2 Instances
+
+Previously, when trying to import an EC2 instance with `pulumi import`, you'd be presented with a list of required properties that weren't satisfied. The challenge here is that none of these fields are required by the schema, but at least one is required for the resource to be valid. Like above, the new behavior will read from the provider and inherit the missing properties, providing everything required to codegen. This provides a much cleaner and intuitive experience for the user.
```

Comment:
This is fairly full with jargon. I don't think most readers know what "schema" or "codegen" are exactly.